Atomic/sign.py

```diff
@@ -72,6 +72,9 @@ def no_reg_no_default_error(image, registry_path):
                     if signature_path is None:
                         raise ValueError("No write path for {}/{} was "
                                          "found in {}".format(reg, repo, registry_config_path))
+                    elif signature_path.startswith("http"):
+                        raise ValueError("Writing to {} is not supported. Use a "
+                                         "file:///location instead.".format(signature_path))
 
                     # Deal with write path prepends
                     if signature_path.startswith("file://"):
```

Comment:
That’s not what I mean.
```python
url =  urlparse.urlparse(signature_path) # + error handling
switch url.scheme:
   case 'file': ... # use url.path
   default: ... # fail with "scheme %s not supported for writing signatures" % url.schem
```

or something; parse using the real structure of the string, do not just check for a small subset of possible values.